### ISSUE TITLE
Introduce extended Query interfaces for CRUD operations

### ISSUE THREAD

**Issue by mecha** [Dhii/storage-interface#6]
## The Problem

Consumers are expected to create query instances using `QueryInterface` as the main contract. The various `*AwareInterface` interfaces are meant to define and standardize the properties that different queries can have.

However, it can be inconvenient for consumers to create and maintain their specific query interfaces and implementations.

## Proposed Solution

Introduce four interfaces that extend [`QueryInterface`](https://github.com/Dhii/storage-interface/blob/master/src/Query/QueryInterface.php).

These would group the appropriate `*AwareInterface` interfaces into singular interfaces that represent the typical properties of a query for each CRUD operation.

In short, to introduce:

* `CreateQueryInterface`
* `ReadQueryInterface`
* `UpdateQueryInterface`
* `DeleteQueryInterface`

This should provide consumers with a set of interfaces that define the typical properties that queries can have. It also allows consumers to rely on these methods for interoperability purposes - even if they are not directly referenced anywhere in this package.
